pkg/util/channel/channel.go

```diff
@@ -20,3 +20,14 @@ func Clear[T any, V chan T | <-chan T](ch V) {
 	for range ch {
 	}
 }
+
+// IsClosed is to check if the channel is closed
+func IsClosed[T any, V chan T | <-chan T](ch V) bool {
+	select {
+	case <-ch:
```

Comment:
This is more general and correct:

```
case _, ok := <-ch:
    return !ok
```

Over all, I dislike introducing additional functions where we can just use language builtin support.

Reply:
Sure, I find it is a bad function. it will remove it.

Reply:
Try to test it:

```
ch := make(chan int)
ch <- 1
util.IsClosed(ch)
```